spec/model/vm_host_spec.rb

```diff
@@ -64,10 +64,10 @@
     expect(vh.ip6_random_vm_network.to_s).not_to eq(vh.ip6_reserved_network)
   end
 
-  it "can generate ipv6 for hosts with smaller than /64 prefix with single byte" do
+  it "can generate ipv6 for hosts with smaller than /64 prefix with two bytes" do
     vh.net6 = NetAddr.parse_net("2a01:4f9:2b:35a::/68")
-    expect(SecureRandom).to receive(:random_number).with(2...256).and_return(5)
-    expect(vh.ip6_random_vm_network.to_s).to eq("2a01:4f9:2b:35a:4::/79")
+    expect(SecureRandom).to receive(:random_number).with(2...2**16).and_return(5)
+    expect(vh.ip6_random_vm_network.to_s).to eq("2a01:4f9:2b:35a:0:4000:0:0/83")
```

Comment:
I would expand this test to do some boundary conditions: the lowest value, the highest value. I guess a middle value for good measure, and freeze the result of that bit-bashing into the test.

Reply:
I might also consider an all-zero address as the test fixture, to make sure the boundaries of your shifting are right, since a bitwise-or will make ambiguous whether a set bit is from the test fixture (is it set from the get-go, or did the `|` set it?)

Reply:
Added new tests